File: lib/test/t_InverseNormalFactory_std.cxx
//                                               -*- C++ -*-
/**
 *  @file  t_InverseNormalFactory_std.cxx
 *  @brief The test file of class InverseNormal for standard methods
 *
 *  Copyright (C) 2005-2013 EDF-EADS-Phimeca
 *
 *  This library is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU Lesser General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This library is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU Lesser General Public License for more details.
 *
 *  You should have received a copy of the GNU Lesser General Public
 *  along with this library.  If not, see <http://www.gnu.org/licenses/>.
 *
 *  @author schueller
 *  @date   2011-04-11 12:32:27 +0200 (lun. 11 avril 2011)
 */
#include "OT.hxx"
#include "OTtestcode.hxx"

using namespace OT;
using namespace OT::Test;

int main(int argc, char *argv[])
{
  TESTPREAMBLE;
  OStream fullprint(std::cout);
  setRandomGenerator();

  try
    {
      InverseNormal distribution(2.5, 1.5);
      UnsignedLong size(10000);
      NumericalSample sample(distribution.getSample(size));
      InverseNormalFactory factory;
      CovarianceMatrix covariance;
      // Distribution estimatedDistribution(factory.build(sample, covariance));
      Distribution estimatedDistribution(factory.build(sample));
      fullprint << "Distribution          =" << distribution << std::endl;
      fullprint << "Estimated distribution=" << estimatedDistribution << std::endl;
      // fullprint << "Covariance=" << covariance << std::endl;
      estimatedDistribution = factory.build();
      fullprint << "Default distribution=" << estimatedDistribution << std::endl;
      estimatedDistribution = factory.build(distribution.getParametersCollection());
      fullprint << "Distribution from parameters=" << estimatedDistribution << std::endl;
      InverseNormal estimatedInverseNormal(factory.buildAsInverseNormal(sample));
      fullprint << "InverseNormal          =" << distribution << std::endl;
      fullprint << "Estimated inverseNormal=" << estimatedInverseNormal << std::endl;
      estimatedInverseNormal = factory.buildAsInverseNormal();
      fullprint << "Default inverseNormal=" << estimatedInverseNormal << std::endl;
      estimatedInverseNormal = factory.buildAsInverseNormal(distribution.getParametersCollection());
      fullprint << "InverseNormal from parameters=" << estimatedInverseNormal << std::endl;
    }
  catch (TestFailed & ex)
    {
      std::cerr << ex << std::endl;
      return ExitCode::Error;
    }


  return ExitCode::Success;
}

File: python/test/t_InverseNormalFactory_std.py
#! /usr/bin/env python

from openturns import *

TESTPREAMBLE()
RandomGenerator().SetSeed(0)

try :
    distribution = InverseNormal(2.5, 1.5)
    size = 10000
    sample = distribution.getSample(size)
    factory = InverseNormalFactory()
    estimatedDistribution = factory.build(sample)
    print "distribution=", repr(distribution)
    print "Estimated distribution=", repr(estimatedDistribution)
    estimatedDistribution = factory.build()
    print "Default distribution=", estimatedDistribution
    estimatedDistribution = factory.build(distribution.getParametersCollection())
    print "Distribution from parameters=", estimatedDistribution
    estimatedInverseNormal = factory.buildAsInverseNormal(sample)
    print "InverseNormal          =", distribution
    print "Estimated inverseNormal=", estimatedInverseNormal
    estimatedInverseNormal = factory.buildAsInverseNormal()
    print "Default inverseNormal=", estimatedInverseNormal
    estimatedInverseNormal = factory.buildAsInverseNormal(distribution.getParametersCollection())
    print "InverseNormal from parameters=", estimatedInverseNormal

except :
    import sys
    print "t_InverseNormalFactory_std.py", sys.exc_type, sys.exc_value
